ws-client/src/transport.rs

```diff
@@ -201,8 +204,9 @@ impl<'a> WsTransportClientBuilder<'a> {
 
 	/// Try establish the connection.
 	pub async fn build(self) -> Result<(Sender, Receiver), WsHandshakeError> {
+		let mut client_config = None;
```

Comment:
what I meant is that you could do:

```rust
let client_config = match self.client_config {
   Some(client_config) => client_config,
   None => {
      let mut config = rustls::ClientConfig::default();
	  if self.use_system_certificates {
	    config.root_store = rustls_native_certs::load_native_certs().map_err(|(_, e)| e)?;
	}
	client_config.get_or_insert(config)
	}
};
let connector: async_tls::TlsConnector = client_config.into();
```

here instead and just pass by the connector by reference to `try_connect`?!

Reply:
Yes, i did not realize we have access to self.mode in `fn build`. Sorry :)